daisy_workflows/image_import/suse/translate.py

```diff
@@ -110,19 +112,85 @@ def _get_distro(g) -> _SuseRelease:
       g.gcp_image_distro, g.gcp_image_major, g.gcp_image_minor))
 
 
-def _install_subscriptions(distro, g):
-  """Executes SuseConnect -p for each subscription on `distro`.
+def _disambiguate_suseconnect_product_error(distro, g, product,
+    error) -> Exception:
+  """Creates a user-debuggable error after failing to add a product
+     using SUSEConnect.
+
+  Args:
+      distro (_SuseRelease): The distro detected by GuestFS.
+      g (GuestFS): Mounted GuestFS instance.
+      product (str): The product that failed to be added.
+      error (Exception): The error returned from `SUSEConnect -p`.
+  """
+  statuses = []
+  try:
+    statuses = json.loads(g.command(['SUSEConnect', '--status']))
+  except Exception as e:
+    return ValueError(
+        'Unable to communicate with SCC. Ensure the import '
+        'is running in a network that allows internet access.', e)
+
+  # `SUSEConnect --status` returns a list of status objects,
+  # where the triple of (identifier, version, arch) uniquely
+  # identifies a product in SCC. Below are two examples.
+  #
+  # Example 1: SLES for SAP 12.2, No subscription
+  # [
+  #    {
+  #       "identifier":"SLES_SAP",
+  #       "version":"12.2",
+  #       "arch":"x86_64",
+  #       "status":"Not Registered"
+  #    }
+  # ]
+  #
+  # Example 2: SLES 15.1, Active
+  # [
+  #    {
+  #       "type":"evaluation",
+  #       "starts_at":"2019-12-16 00:00:00 UTC",
+  #       "status":"Registered",
+  #       "version":"15.1",
+  #       "expires_at":"2020-02-14 00:00:00 UTC",
+  #       "arch":"x86_64",
+  #       "identifier":"SLES",
+  #       "subscription_status":"ACTIVE"
+  #    },
+  #    {
+  #       "status":"Registered",
+  #       "version":"15.1",
+  #       "arch":"x86_64",
+  #       "identifier":"sle-module-basesystem"
+  #    }
+  # ]
+
+  for status in statuses:
+    if status.get('identifier') not in ('SLES', 'SLES_SAP'):
+      continue
+
+    if status.get('subscription_status') is 'ACTIVE':
```

Comment:
In which cases an error is returned for an active subscription?

Reply:
It's one of those "should never happen" scenarios, as far as I know, and by that I mean: right now we think it won't happen, but over time, that code will get executed.